### ISSUE TITLE
ffi:Pointer finalizer

### ISSUE THREAD

**Issue by Buggaboo** [objectbox/objectbox-dart#53]
We should track the development of [this issue](https://github.com/dart-lang/sdk/issues/35770).
It might be available on the bleeding edge dart.

**Comment by Sunbreak** [comment]
https://github.com/dart-lang/sdk/issues/35770#issuecomment-650158676

dart:ffi supports GC finalizers now!

**Comment by vaind** [comment]
The way this can be [used already doesn't look so bad](https://github.com/dart-lang/sdk/issues/35770#issuecomment-766717146). If we could have a single obx_dart_register_finalizer() in the native API, that wouldn't be too bad I guess... At least we could get rid of all the close() calls and implement them properly once finalizers land in user-space dart. Let's consider for 1.0

**Comment by Buggaboo** [comment]
Structs / Pointers are disallowed finalizers, due to potential [use-after-freeing memory](https://github.com/dart-lang/sdk/issues/45071)

**Comment by vaind** [comment]
I guess we're gonna wait how this turns out. With pointers out of the question, we wouldn't be able to use finalizers for what we need them.

**Comment by vaind** [comment]
Another follow-up issue in dart SDK: https://github.com/dart-lang/sdk/issues/45455

**Comment by vaind** [comment]
I've ended up implementing this using the currently available finalizers and it works fine.
